src/main/java/wooteco/subway/service/LineService.java

```diff
@@ -0,0 +1,61 @@
+package wooteco.subway.service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import wooteco.subway.dao.JdbcLineDao;
+import wooteco.subway.domain.Line;
+import wooteco.subway.dto.LineRequest;
+import wooteco.subway.dto.LineResponse;
+import wooteco.subway.dto.LineUpdateRequest;
+import wooteco.subway.exception.BadRequestException;
+import wooteco.subway.exception.NotFoundException;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private final JdbcLineDao jdbcLineDao;
+
+    public LineService(JdbcLineDao jdbcLineDao) {
+        this.jdbcLineDao = jdbcLineDao;
+    }
+
+    @Transactional
+    public LineResponse create(LineRequest lineRequest) {
+        Line line = new Line(lineRequest.getName(), lineRequest.getColor());
+        validateDuplicateNameAndColor(line.getName(), line.getColor());
+        return LineResponse.from(jdbcLineDao.save(line));
+    }
+
+    public LineResponse showById(Long id) {
+        return LineResponse.from(
+            jdbcLineDao.findById(id)
+                .orElseThrow(() -> new NotFoundException("조회하려는 id가 존재하지 않습니다."))
+        );
```

Comment:
```suggestion
        Line line = jdbcLineDao.findById(id)
                .orElseThrow(() -> new NotFoundException("조회하려는 id가 존재하지 않습니다."))
        return LineResponse.from(line);
```

저도 가끔 코딩하다보면 인라이닝에 집착하게 되는 때가 있는데, 그냥 두 줄로 표현하는 게 훨씬 보기 좋을 때가 많더라구요 😁

Reply:
갑자기 사이다를 마시게 된 기분이네요. 여러번 수정해보면서 어떤게 잘 읽힐까 고민했는데 아에 분리하는 것도 방법이네요👍👍